### PR TITLE
IEP-525 Show Application RAM and flash usage info after the build

### PR DESCRIPTION
**Pull request by sigmaaa**
Now, after the build, we can see the application's RAM and flash usage using the idf_size.py command:

Screenshot 2022-01-05 at 20 56 50

If the remaining app size is less than 30%, we will show the warning message with the link to the [documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html?highlight=partitions%20csv#creating-custom-tables)

Screenshot 2022-01-05 at 20 56 20

### CODE REVIEW

**Comment by sigmaaa** [comment]
Hi @igrr, good point. I'll add that. Thanks!

**Comment by sigmaaa** [comment]
Hi, @igrr. After the latest commit, the message dialog looks like this:
Screenshot 2022-01-06 at 0 09 39

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
Thanks @sigmaaa for the PR. LGTM. A couple of other things we could consider - 

- Refactor code -  Move all partition size checking code into the `ParitionSizeHandler` class and invoke it from the IDFBuildConfiguration. This includes checkRemainingSize, getPartitionTable, startIdfSizeProcess and getMapFilePath() etc.
- When we are showing a warning message to the user, maybe we could be very specific on the size details. 

For example, I could suggest something like this `Less than 30% of application partition size is free(X of Y MB), would you like to increase it? Please click here to check more details.`

File: bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/MessageLinkDialog.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+		});
+		checkBox.setText(Messages.MsgLinkDialog_DoNotShowMsg);
+		return parent;
+	}
+
+	public static void openWarning(Shell parent, String title, String message)
+	{
+		preferences = InstanceScope.INSTANCE.getNode("com.espressif.idf.launch.serial.ui"); //$NON-NLS-1$
+		if (!preferences.getBoolean(DO_NOT_SHOW_MSG, false))
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/MessageLinkDialog.java]: Thanks, @kolipakakondal for the review, added the requested changes. Now message looks like this:
Screenshot 2022-01-06 at 16 33 09

File: bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/messages.properties [hunk 2]
```diff
@@ -15,3 +15,5 @@ CMakeBuildConfiguration_ProcCompJson=Processing compile_commands.json
 CMakeBuildConfiguration_Failure=Failure running cmake: %s\n
 CMakeErrorParser_NotAWorkspaceResource=Could not map %s to a workspace resource. Did the build run in a container?
 IDFBuildConfiguration_CMakeBuildConfiguration_NoToolchainFile=No toolchain file found
+IncreasePartitionSizeTitle=Low Application Partition Size
+IncreasePartitionSizeMessage=Consider increasing the application partition size, see <a href=\"https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html?highlight=partitions%20csv#creating-custom-tables\">link</a> for details.
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/messages.properties]: done

File: bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/MessageLinkDialog.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Shell;
+
+public class MessageLinkDialog extends MessageDialog
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/MessageLinkDialog.java]: done

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**